src/lib.rs

```diff
@@ -328,23 +328,25 @@ pub fn generate_flamegraph_for_workload(workload: Workload, opts: Options) -> an
         signal_hook::low_level::register(SIGINT, || {}).expect("cannot register signal handler")
     };
 
-    let perf_output = if let Workload::ReadPerf(perf_file) = workload {
-        Some(perf_file)
-    } else {
-        arch::initial_command(
-            workload,
-            opts.root,
-            opts.frequency,
-            opts.custom_cmd,
-            opts.verbose,
-            opts.ignore_status,
-        )
-    };
-
+    for _i in 0..iterations {
+        if let Workload::ReadPerf(_perf_file) = &workload {
+            // pass
```

Comment:
Did you test this with a `Workload::ReadPerf` scenario? I guess maybe this `if` needs to be outside this loop, so that we can still pass `perf_file` to `output()` (as `perf_output`)?